app/configuration.py

```diff
@@ -55,6 +58,8 @@ class MqttSettings:
 
     def __post_init__(self):
         self.pub_errors = self.error_topic is not None and len(self.error_topic) > 0
+        if self.pub_errors:
+            logging.info(f"Publishing errors under {self.error_topic}")
```

Comment:
I don't mind have this logging, but It feels a bit odd how reads.
This class is about parsing a configuration file.
`Publishing` reads more behavioural. I wonder if we could rephrase this in a way that reads closer to "configured to publish on..."

Reply:
In other words, the gerund would probably be better suited to be in the `MQTT writer` instead of this file.

Reply:
Yes it probably should say "Configured to publish errors under ..." - it's related to the initial configuration parsing rather than a runtime thing. Probably it would be better in the ErrorHandler init

Reply:
Yes, the ErrorHanlder init would be a good candidate to place the gerund form of the verb.
But nothing wrong to have a logging message here. The one you have listed above is a better form for this class.

Reply:
Upon instantiation of the file, you can do a full replacement of environment variables on the the entire document.
You scan all values. Anytime you see `${THING}` you do replacement or fail/exception.

After that, each component or part, do not need to be doing manually any env replacement. You only care about selecting values.

By doing that separation, you separate mutation from extraction.